pilot/pkg/networking/plugin/authz/rbac.go

```diff
@@ -199,18 +197,53 @@ func convertRbacRulesToFilterConfig(service string, roles []model.Config, bindin
 				permissions = append(permissions, convertToPermission(rule))
 			}
 		}
+		if len(permissions) == 0 {
+			log.Debugf("role %v skipped for no permissions found", role.Name)
+			continue
+		}
 
-		if len(permissions) != 0 {
+		enforcedPrincipals, permissivePrincipals := convertToPrincipals(roleToBinding[role.Name])
+		principals := make([]*policyproto.Principal, 0)
+		principals = append(principals, enforcedPrincipals...)
+		principals = append(principals, permissivePrincipals...)
+
+		if len(principals) == 0 {
+			// Skip to next role if found no bindings for current role. This means nobody could
+			// access the current role, so we don't need to check the remaining rules anymore.
+			log.Debugf("role %v skipped for no bindings found", role.Name)
+			continue
+		}
+
+		// If ServiceRole is in PERMISSIVE mode, all ServiceRoleBindings that refers to it will be treated as PERMISSIVE mode.
+		if role.Spec.(*rbacproto.ServiceRole).Mode == rbacproto.EnforcementMode_PERMISSIVE {
 			// Constructs the policy for a role with both permissions and bindings.
-			rbac.Policies[role.Name] = &policyproto.Policy{
+			permissiveRbac.Policies[role.Name] = &policyproto.Policy{
 				Permissions: permissions,
 				Principals:  principals,
 			}
-			log.Debugf("role %v generated policy: %v", role.Name, *rbac.Policies[role.Name])
+			log.Debugf("role %v generated permissive policy: %v", role.Name, *permissiveRbac.Policies[role.Name])
+
+			continue
```

Comment:
Please use `else` instead of `continue` to handle the ENFORCED case.

Reply:
done.